modules/azure/storage.go

```diff
@@ -0,0 +1,235 @@
+	if err != nil {
+		return "", err
+	}
+	return string(storageAccount.Kind), nil
+}
+
+// GetStorageAccountSkuTierE returns the storage account sku tier as Standard or Premium.
+func GetStorageAccountSkuTierE(storageAccountName, resourceGroupName, subscriptionID string) (string, error) {
+	storageAccount, err := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	return string(storageAccount.Sku.Tier), nil
+}
+
+// GetStorageBlobContainerE returns Blob container client.
+func GetStorageBlobContainerE(containerName, storageAccountName, resourceGroupName, subscriptionID string) (*storage.BlobContainer, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageBlobContainerClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	container, err := client.Get(context.Background(), resourceGroupName, storageAccountName, containerName)
+	if err != nil {
+		return nil, err
+	}
+	return &container, nil
+}
+
+// GetStorageAccountPropertyE returns StorageAccount properties.
+func GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID string) (*storage.Account, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageAccountClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	account, err := client.GetProperties(context.Background(), resourceGroupName, storageAccountName, "")
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
+}
+
+// GetStorageAccountClientE creates a storage account client.
+func GetStorageAccountClientE(subscriptionID string) (*storage.AccountsClient, error) {
+	storageAccountClient := storage.NewAccountsClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+	storageAccountClient.Authorizer = *authorizer
+	return &storageAccountClient, nil
+}
+
+// GetStorageBlobContainerClientE creates a storage container client.
+func GetStorageBlobContainerClientE(subscriptionID string) (*storage.BlobContainersClient, error) {
+	blobContainerClient := storage.NewBlobContainersClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+	blobContainerClient.Authorizer = *authorizer
+	return &blobContainerClient, nil
+}
+
+// GetStorageURISuffixE returns the proper storage URI suffix for the configured Azure environment.
+func GetStorageURISuffixE() (*string, error) {
+	var envName = "AzurePublicCloud"
```

Comment:
Idiomatic way to declare and assign var in golang is to use `:=`
```suggestion
	envName := "AzurePublicCloud"
```